graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/nodes/promise/ImportCallNode.java

```diff
@@ -109,6 +118,25 @@ public Object execute(VirtualFrame frame) {
         Object referencingScriptOrModule = getActiveScriptOrModule(frame);
         Object specifier = argRefNode.execute(frame);
         String specifierString;
+
+        if (specifier instanceof DynamicObject && DynamicObjectLibrary.getUncached().containsKey((DynamicObject) specifier, (Object) ModuleBlockNode.getModuleBodyKey())) {
```

Comment:
Is there a specific reason you are using DynamicObjectLibrary here? I believe `JSObject.hasProperty(...)` is what you should use here.

Reply:
Used JSObjectUtil.hasHiddenProperty(...) and applied it at file ModuleBlockPrototypeBuiltins as well.